packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx

```diff
@@ -35,92 +29,31 @@ const LISTEN_OPTIONS: ListenOptions = {
 export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPerspectiveState {
   const {documentId} = props
 
-  const [state, setState] = useState<BundleDocument[] | null>(null)
-
-  const [error, setError] = useState<Error | undefined>(undefined)
   const client = useClient(DEFAULT_STUDIO_CLIENT_OPTIONS)
   const {data: bundles} = useBundles()
   const publishedId = getPublishedId(documentId, documentId.includes('.'))
 
-  const QUERY = `*[sanity::versionOf("${publishedId}")]`
-
-  const initialFetch = useCallback(async () => {
-    if (!client) {
-      return
-    }
-
-    try {
-      const res = await client.fetch(QUERY)
-
-      // build the document bundles
-      const documentBundles = res
-        ?.map((r: BundleDocument) =>
-          bundles?.find((b) => r._version && getBundleSlug(r._id) === b.slug),
-        )
-        .filter((b: BundleDocument) => b !== undefined)
-
-      setState(documentBundles)
-    } catch (err) {
-      setError(err)
-    }
-  }, [QUERY, bundles, client])
+  const QUERY = `*[sanity::versionOf("${publishedId}")]{_id, _version}`
```

Comment:
Q: need we project to include `_version` when all these docs will have `_version: {}`?

Reply:
Ah, all good, I see that versionOf also returns draft and published so `_version` is used to find only bundles